### PR TITLE
[BOOST-5790] add tuple support for claimant in sdk

### PR DESCRIPTION
**Pull request by mmackz**
### Description
- add support for single level tuples for claimant

## Summary by CodeRabbit

- **New Features**
  - Claimant resolution now supports nested tuple fields in events and function arguments.
  - Tuple index encoding added to target nested values while keeping single-index behavior.
  - Ability to extract scalar values from nested tuples for incentive criteria.
  - Improved validation and safeguards for invalid or out-of-bounds tuple access.

- **Tests**
  - Expanded coverage for tuple-based claimant access, edge cases, and mixed structures.

- **Chores**
  - Prepared a minor SDK release.

### CODE REVIEW

**Comment by jonathandiep** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Please add some non-placeholder details to the PR description.</td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against bf950a59b0fd86d795731c072ef85a4cc2d4dd29
</p>

File: packages/sdk/src/Actions/EventAction.ts [hunk 1]
```diff
@@ -1990,6 +2007,36 @@ export function isCriteriaFieldIndexTuple(fieldIndex: number): boolean {
   return fieldIndex >= 32;
 }
 
+/**
+ * Helper function to check if a claimant fieldIndex represents tuple access.
```
Comment (mmackz) [inline packages/sdk/src/Actions/EventAction.ts:2011]: created wrappers around the incentive criteria helpers to make it more semantic.
